docker.go

```diff
@@ -90,6 +98,36 @@ func (container *Container) EmptyVolumes() *Container {
 	return container
 }
 
+// Define external volumes attached to the container
+//      containerPath: the path inside the container to map the volume
+//      name: the name of the volume
+//      provider: the provider of the volume (e.g. dvdi)
+//      mode: the read/write mode for the volume (e.g. RW)
+//      opts: the map of options for the external volume
+func (container *Container) ExternalVolume(containerPath string,
```

Comment:
- This duplicates a bit of functionality from [Volume()](https://github.com/bosschaert/go-marathon/blob/18b68aa3267107dc7ea9e6bbfc3f77b659273906/docker.go#L76). I think `ExternalVolume` should only receive parameters specific to an external volume (i.e., `name`, `provider`, and `opts`) and just create an `ExternalVolume`. Users would then call the method, assign the result to a `Volume` instance, and add that volume to the container.
- You can reduce the consecutive string parameter declarations by doing `name, provider, mode string`.
- `opts` is a bit unwieldy if you don't need it as you'd have to pass in an empty map like `map[string]string{}`. Plus, having this parameter conceptually means we'd need to extend the signature in a breaking way for every new parameter we get in the future. How about attaching a method `SetOpts()` to `ExternalVolume` that takes a map instead? If we pair it with an `EmptyOpts()` method, we also allow users to clear options.